File: SnackDown'19/Chef_And_Typing.py
def main() :
    t = int(input())
    left = 'df'
    right = 'jk'
    while t :
        t = t - 1
        total = 0
        words = {}
        n = int(input())
        for i in range(n) :
            time = 0
            word = input()
            if word not in words :
                time = 0.2
                words[word] = 0
                for y in range(1,len(word)) :
                    if word[y-1] in left and word[y] in right :
                        time += 0.2
                    elif word[y-1] in right and word[y] in left :
                        time += 0.2
                    else :
                        time += 0.4
                words[word] = time 
            else :
                time = words[word]/2
            total += time
        print(int(total*10))
main()

File: SnackDown'19/Chef_And_Typing.cpp
#include <iostream>
#include <vector>
#include <algorithm> 
#include <string> 

using namespace std;

int main() {
    
    int t;
    cin >> t ;
    while( t-- ) {
        
        int n;
        cin >> n;
        vector< string > v(n), left(2), right(2) ;
        left[0] = "d" ; left[1] = "f" ; 
        right[0] = "j" ; right[1] = "k" ; 
        cout << endl ;
        for ( auto c : left ) cout << c <<" " ;
        cout << endl ;
        for ( auto c : right ) cout << c <<" " ;
        for ( auto i = 0 ; i < n ; ++i ) cin >> v[i] ;
        




    
    }


}